### PR TITLE
Experimental code to increase robustness of cubic params

### PR DESCRIPTION
**Pull request by raphlinus**
This only calls atan2 when the derivatives are sufficiently large.

WIP, not to commit, as GPU and CPU are out of sync.

### CODE REVIEW

File: shader/flatten.wgsl [hunk 1]
```diff
@@ -200,7 +217,7 @@ fn es_seg_from_params(p0: vec2f, p1: vec2f, params: EulerParams) -> EulerSeg {
 
 fn es_seg_eval_with_offset(es: EulerSeg, t: f32, offset: f32) -> vec2f {
     let chord = es.p1 - es.p0;
-    let scaled = offset / length(chord);
+    let scaled = offset / max(length(chord), TANGENT_THRESH);
```
Comment (armansito) [inline shader/flatten.wgsl:220]: Not sure if this should be capped at `sqrt(TANGENT_THRESH)` instead of `TANGENT_THRES`.

**Comment by armansito** [comment]
I pushed commits with a fix for divides by zero chord length, a minor clean up of constants, and a port of the fixes to the CPU euler code.

Seeing as the AMD issues likely have a different root cause, I think we can move this PR out of draft and land it while investigating the other issues.

**Comment by armansito** [comment]
This is superseded by #521, which incorporates the fixes from here plus some restructuring that makes the code cleaner.

**Comment by raphlinus** [comment]
Indeed, closing in favor of #521.

**Pull request closed without merging**